### PR TITLE
Fix test crashes on dropping indexer runtime

### PR DESCRIPTION
**Pull request by mzabaluev**
# Summary
- **Categories**: `protocol-units`, `cicd`.

Fixes: #573

# Changelog

* Revert `Executor::background` to previous style returning two objects. The `IndexerRuntime` can be spun with a separate method.

# Testing

`cargo test -p maptos-opt-executor`

Now exercised in CI 🙌 

`cargo test -p maptos-fin-view` (currently requires the Movement env)

# Outstanding issues

Tests in `maptos-fin-view` remain to be adapted as per #553

### CODE REVIEW

**Review by l-monninger — CHANGES_REQUESTED** [review]
CI appears to be hanging.

**Comment by mzabaluev** [comment]
> CI appears to be hanging.

Odd, it's only one run in the matrix. I'll check if a merge with newer commits in main (which is what the workflow triggered by `pull_request` checks out) is causing this.

BTW, should we restrict the `push` trigger to well-known branches (currently 'main')? Now the checks are run largely redundantly, except for the difference in checkout as explained above.

**Comment by mzabaluev** [comment]
> CI appears to be hanging.

In main, there have only been Solidity changes, this should not affect the unit tests.

Looks like we've hit a random deadlock.

**Comment by mzabaluev** [comment]
```
```

**Comment by mzabaluev** [comment]
The cause of the lockup is, as seen in the lower stack frame, dropping a registry entry results in a recursion where a referenced resource is removed from the registry while the lock is already held on its registry shard.

I've been trying to determine how likely is this recursion to occur in normal operation. In my testing on Mac, it occurs once after several dozen iterations. On a PC in CI it might be easier to trigger.

**Comment by mzabaluev** [comment]
Tests fixed by https://github.com/movementlabsxyz/aptos-core/pull/67

**Review by l-monninger — APPROVED** [review]

File: .github/workflows/checks-all.yml [hunk 1]
```diff
@@ -129,8 +156,8 @@ jobs:
     - name: Run Suzuka Full Node Tests Against Holesky and Local Celestia
       env: 
         CELESTIA_LOG_LEVEL: FATAL # adjust the log level while debugging
+        MCR_DEPLOYMENT_ACCOUNT_PRIVATE_KEY: ${{ secrets.MCR_DEPLOYMENT_ACCOUNT_PRIVATE_KEY }}
```
Comment (l-monninger) [inline .github/workflows/checks-all.yml:159]: Why did you switch this from using the generated key?

Comment (mzabaluev) [reply .github/workflows/checks-all.yml:159]: No switch, just moved this from the run script where the secret could leak into the logs (more easily) or `ps` output.

**Pull request merged**